// [[Reason: Function declarations are not fixed]] ////

//// [tests/cases/compiler/isolatedDeclarationErrors.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -2,9 +2,9 @@
 
 //// [isolatedDeclarationErrors.d.ts]
 declare function errorOnAssignmentBelowDecl(): void;
 declare namespace errorOnAssignmentBelowDecl {
-    var a: string;
+    var a: invalid;
 }
 declare const errorOnAssignmentBelow: {
     (): void;
     a: string;
@@ -12,5 +12,28 @@
 declare const errorOnMissingReturn: {
     (): void;
     a: string;
 };
-//# sourceMappingURL=isolatedDeclarationErrors.d.ts.map
\ No newline at end of file
+//# sourceMappingURL=isolatedDeclarationErrors.d.ts.map
+/// [Errors] ////
+
+isolatedDeclarationErrors.ts(2,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+
+
+==== isolatedDeclarationErrors.ts (1 errors) ====
+    function errorOnAssignmentBelowDecl(): void {}
+    errorOnAssignmentBelowDecl.a = "";
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    
+    const errorOnAssignmentBelow: {
+        (): void;
+        a: string;
+    } = (): void => {}
+    errorOnAssignmentBelow.a = "";
+    
+    const errorOnMissingReturn: {
+        (): void;
+        a: string;
+    } = () => {}
+    errorOnMissingReturn.a = "";
+    
\ No newline at end of file
